adagucserverEC/CDrawImage.cpp

```diff
@@ -866,28 +884,29 @@ void CDrawImage::setText(const char *text, size_t length, int x, int y, CColor c
   }
 }
 
-void CDrawImage::setTextStroke(const char *text, size_t length, int x, int y, int fgcolor, int bgcolor, int fontSize) {
+void CDrawImage::setTextStroke(int x, int y, float angle, const char *text, const char *fontFile, float fontSize, float strokeWidth, CColor bgcolor, CColor fgcolor) {
   if (currentGraphicsRenderer == CDRAWIMAGERENDERER_CAIRO) {
-    // Not yet supported...
-    setText(text, length, x, y, fgcolor, fontSize);
+    if (bgcolor.a == 0) {
+      drawText(x, y, fontFile, fontSize, angle, text, fgcolor);
+    } else {
+      cairo->drawStrokedText(x, y, -angle, text, fontFile, fontSize * 1.4, strokeWidth, bgcolor, fgcolor);
```

Comment:
Angle is defined different in local drawText versus cairo->drawStrokedText?

Reply:
Yes, drawText uses freetype and drawstroked text uses cairo...